addons/medical_vitals/functions/fnc_updateOxygen.sqf

```diff
@@ -0,0 +1,81 @@
+#include "script_component.hpp"
+/*
+ * Author: Brett Mayson
+ * Update the oxygen levels
+ *
+ * Arguments:
+ * 0: The Unit <OBJECT>
+ * 1: Time since last update <NUMBER>
+ * 2: Sync value? <BOOL>
+ *
+ * ReturnValue:
+ * Current SPO2 <NUMBER>
+ *
+ * Example:
+ * [player, 1, false] call ace_medical_vitals_fnc_updateOxygen
+ *
+ * Public: No
+ */
+
+params ["_unit", "_deltaT", "_syncValue"];
+
+#define IDEAL_PPO2 0.255
+
+private _current = GET_SPO2(_unit);
+private _heartRate = GET_HEART_RATE(_unit);
+
+private _altitude = EGVAR(common,mapAltitude) + ((getPosASL _unit) select 2);
+private _po2 = if (missionNamespace getVariable ["ace_weather", false]) then {
+    private _temperature = _altitude call EFUNC(weather,calculateTemperatureAtHeight);
+    private _pressure = _altitude call EFUNC(weather,calculateBarometricPressure);
+    [_temperature, _pressure, EGVAR(weather,currentHumidity)] call EFUNC(weather,calculateOxygenDensity)
+} else {
+    // Rough approximation of the partial pressure of oxygen in the air
+    0.25725 * (_altitude / 1000 + 1)
+};
+
+private _oxygenSaturation = (IDEAL_PPO2 min _po2) / IDEAL_PPO2;
+
+// Check gear for oxygen supply
+[goggles _unit, headgear _unit] findIf {
+    if (_x != "") then {
+        if (isNil QGVAR(oxygenSupplyConditionCache)) then {
+            GVAR(oxygenSupplyConditionCache) = createHashmap;
+        };
+        private _condition = GVAR(oxygenSupplyConditionCache) getOrDefaultCall [_x, {
+            compile getText (configFile >> "CfgWeapons" >> _x >> QGVAR(oxygenSupply))
+        }, true];
+        if (_condition isNotEqualTo {} && {ACE_player call _condition}) then {
+            _oxygenSaturation = 1;
+            _po2 = IDEAL_PPO2;
+            true
+        } else {
+            false
+        }
+    } else {
+        false
+    }
+};
```

Comment:
Can the `_condition` lookup/compile be moved to preStart and then just do a simple `_x in QGVAR(oxygenSupplyConditionCache)` here? This block looks ugly in the middle of the function. Either that or move it into a separate file.